packages/block-library/src/post-terms/index.php

```diff
@@ -36,12 +36,22 @@ function render_block_core_post_terms( $attributes, $content, $block ) {
 
 	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $classes ) );
 
+	$prefix = "<div $wrapper_attributes>";
+	if ( isset( $attributes['prefix'] ) && $attributes['prefix'] ) {
+		$prefix .= '<span class="wp-block-post-terms__prefix">' . $attributes['prefix'] . '</span>';
```

Comment:
Thank you for working on this. Tested the functionality and I can see prefix and suffix working. I'm afraid we should escape the content of both `prefix` and `suffix` before passing it inside `get_the_term_list `.  Let me know what do you think?

Reply:
I don't think it should be escaped, but I could very well be wrong. 

Adding some HTML i.e. that which `InnerText` has toolbar buttons for, is a feature, and stripping it would lead to a weird editing experience - what you see in the editor is very different to what you get when viewing the site.

Additionally I'm not aware of what security problem it poses over just using e.g. the paragraph block and the code editor view to add:
```
<!-- wp:paragraph -->
<p><script>alert('hello');</script></p>
<!-- /wp:paragraph -->
```

So far as I know WordPress is reliant on `wp_kses_post` when saving to validate the content is ok, and that might strip javascript from the POST, depending upon the type of account you have on that site.


That said, stripping all HTML is the approach taken by e.g. the post-author block:
https://github.com/WordPress/gutenberg/blob/c5b9ca14e68ad6ee2aefa2973589f36cac388a7e/packages/block-library/src/post-author/index.php#L42

but that easily leads to output like this when viewing the site, so I'm not sure it's a good role model.
![image](https://user-images.githubusercontent.com/93301/165187328-4657e3c4-cc7c-4f6d-8383-6354d89357cf.png)

Reply:
Filed #40600 to deal with this in the post author block, either way, it's currently a confusing user experience on that block.

Reply:
I agree that it doesn't need to be escaped here between the span tags.